plugins/Divera/Divera.py

```diff
@@ -0,0 +1,153 @@
+    @requires:  nothing
+    @return:    nothing
+    """
+    # nothing to do for this plugin
+    return
+
+
+##
+#
+# Main function of Divera-plugin
+# will be called by the alarmHandler
+#
+def run(typ, freq, data):
+    """
+    This function is the implementation of the Divera-Plugin.
+    It will send the data to Divera API
+    @type    typ:  string (FMS|ZVEI|POC)
+    @param   typ:  Typ of the dataset
+    @type    data: map of data (structure see readme.md in plugin folder)
+    @param   data: Contains the parameter
+    @type    freq: string
+    @keyword freq: frequency of the SDR Stick
+    @requires:  Divera-Configuration has to be set in the config.ini
+    @return:    nothing
+    """
+    try:
+        if configHandler.checkConfig("Divera"):  # read and debug the config
+
+            if typ == "FMS":
+                #
+                # building message for FMS
+                #
+                text = globalVars.config.get("Divera", "fms_text")
+                title = globalVars.config.get("Divera", "fms_title")
+                priority = globalVars.config.get("Divera", "fms_prio")
+
+            elif typ == "ZVEI":
+                #
+                # building message for ZVEI
+                #
+                text = globalVars.config.get("Divera", "zvei_text")
+                title = globalVars.config.get("Divera", "zvei_title")
+                priority = globalVars.config.get("Divera","zvei_std_prio")
+
+            elif typ == "POC":
+                #
+                # building message for POC
+                #
+                if data["function"] == '1':
+                    priority = globalVars.config.get("Divera", "SubA")
+                elif data["function"] == '2':
+                    priority = globalVars.config.get("Divera", "SubB")
+                elif data["function"] == '3':
+                    priority = globalVars.config.get("Divera", "SubC")
+                elif data["function"] == '4':
+                    priority = globalVars.config.get("Divera", "SubD")
+                else:
+                    priority = ''
+                        
+                text = globalVars.config.get("Divera", "poc_text")
+                title = globalVars.config.get("Divera", "poc_title")
+
+            else:
+                logging.warning("Invalid type: %s", typ)
+                return
+
+        try:
+            #
+            # Divera-Request
+            #
+            logging.debug("send Divera for %s", typ)
+
+            # replace the wildcards
+            text = wildcardHandler.replaceWildcards(text, data)
+            title = wildcardHandler.replaceWildcards(title, data)
+            
+            # Logging data to send
+            logging.debug("Title   : %s", title)
+            logging.debug("Text    : %s", text)
+            logging.debug("Priority: %s", priority)
+				
+            # check priority value
+            if (priority != 'false') and (priority != 'true'):
```

Comment:
wäre es hier nicht besser ganz am Anfang mit None vor zu initialisieren `priority = None`,
und dann einfach auf `if not priority is None:` zu schauen ob es inzwischen eine Zuweisung gab?

Reply:
Hab grad config doku gesehen - vergiss was ich geschrieben habe - passt so

Reply:
Könnte man machen, aber wenn jemand einen anderen Wert als true oder false angibt, dann passt das nicht mehr mit der Divera-Schnittstelle